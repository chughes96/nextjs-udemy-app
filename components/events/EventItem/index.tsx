import React from 'react';
import Link from 'next/link';
import { EventDetails } from '@/types';
import styles from './EventItem.module.css';

type Props = {
  event: EventDetails;
};

export const EventItem = ({
  event: { title, image, date, location, id },
}: Props) => {
  const formattedDate = new Date(date).toLocaleDateString('en-GB', {
    day: 'numeric',
    month: 'long',
    year: 'numeric',
  });

  const formattedAddress = location.replace(', ', '\n');

  return (
    <li className={styles.item}>
      <img src={`/${image}`} alt={image} />
      <div className={styles.content}>
        <div className={styles.summary}>
          <h2>{title}</h2>
          <div className={styles.date}>
            <time>{formattedDate}</time>
          </div>
          <div className={styles.address}>
            <address>{formattedAddress}</address>
          </div>
        </div>
        <div className={styles.actions}>
          <Link href={`/events/${id}`}>Explore Event</Link>
        </div>
      </div>
    </li>
  );
};
