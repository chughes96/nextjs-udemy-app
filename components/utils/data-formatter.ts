export const formatDate = (date: string) =>
  new Date(date).toLocaleDateString('en-GB', {
    day: 'numeric',
    month: 'long',
    year: 'numeric',
  });

export const formatAddress = (address: string) => address.replaceAll(', ', '\n');
