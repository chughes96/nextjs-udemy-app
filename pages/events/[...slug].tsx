import React from 'react';

type Props = {};

const FilteredEventsPage = (props: Props) => {
  return <div>Filtered Events</div>;
};

export default FilteredEventsPage;
