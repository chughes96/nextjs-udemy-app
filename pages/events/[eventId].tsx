import React from 'react';
import { useRouter } from 'next/router';
import { ParsedUrlQuery } from 'querystring';
import { EventContent } from '@/components/event-detail/EventContent';
import { EventLogistics } from '@/components/event-detail/EventLogistics';
import { EventSummary } from '@/components/event-detail/EventSummary';
import { getEventById } from '@/dummy-data';

interface QueryParams extends ParsedUrlQuery {
  eventId: string;
}

type Props = {};

const EventDetailsPage = (props: Props) => {
  const router = useRouter();

  const { eventId } = router.query as QueryParams;
  const eventDetails = getEventById(eventId);

  if (!eventDetails) return <p>No event found!</p>;

  return (
    <>
      <EventSummary title={eventDetails.title} />
      <EventLogistics eventDetails={eventDetails} />
      <EventContent>
        <p>{eventDetails.description}</p>
      </EventContent>
    </>
  );
};

export default EventDetailsPage;
