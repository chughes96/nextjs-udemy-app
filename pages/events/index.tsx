import React from 'react';

type Props = {};

const EventsPage = (props: Props) => {
  return <div>Events</div>;
};

export default EventsPage;
