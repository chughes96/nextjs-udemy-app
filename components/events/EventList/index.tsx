import { EventDetails } from '@/types';
import React from 'react';
import { EventItem } from '../EventItem';
import styles from './EventList.module.css';

type Props = {
  events: EventDetails[];
};

export const EventList = ({ events }: Props) => {
  return (
    <ul className={styles.list}>
      {events.map((e) => (
        <EventItem key={e.id} event={e} />
      ))}
    </ul>
  );
};
