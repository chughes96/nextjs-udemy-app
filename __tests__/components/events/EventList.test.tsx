import { render, screen } from '@testing-library/react';
import { createEvents } from '@/__tests__/utils/test-data-builder';
import { EventList } from '@/components/events/EventList';
import { EventDetails } from '@/types';
import { getAllEvents } from '@/dummy-data';

describe('<EventList />', () => {
  let testEvents = createEvents(5);

  const renderComponent = (events: EventDetails[]) =>
    render(<EventList events={events} />);

  it('renders correctly and matches the snapshot', () => {
    // Cannot use test data builder reliably for snapshots because of randomised data
    const events = getAllEvents();

    const component = renderComponent(events);

    expect(component).toMatchSnapshot();
  });

  // Could use it.each but this renders component once
  it('renders an EventItem for each event', () => {
    renderComponent(testEvents);

    testEvents.forEach((e) => {
      const title = screen.getByText(e.title);

      expect(title).toBeInTheDocument();
    });
  });
});
