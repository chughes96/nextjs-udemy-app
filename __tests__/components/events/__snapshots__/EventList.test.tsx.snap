// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventList /> renders correctly and matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <img
            alt="images/coding-event.jpg"
            src="/images/coding-event.jpg"
          />
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <h2>
                Programming for everyone
              </h2>
              <div
                class="date"
              >
                <time>
                  12 May 2021
                </time>
              </div>
              <div
                class="address"
              >
                <address>
                  Somestreet 25
12345 San Somewhereo
                </address>
              </div>
            </div>
            <div
              class="actions"
            >
              <a
                href="/events/e1"
              >
                Explore Event
              </a>
            </div>
          </div>
        </li>
        <li
          class="item"
        >
          <img
            alt="images/introvert-event.jpg"
            src="/images/introvert-event.jpg"
          />
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <h2>
                Networking for introverts
              </h2>
              <div
                class="date"
              >
                <time>
                  30 May 2021
                </time>
              </div>
              <div
                class="address"
              >
                <address>
                  New Wall Street 5
98765 New Work
                </address>
              </div>
            </div>
            <div
              class="actions"
            >
              <a
                href="/events/e2"
              >
                Explore Event
              </a>
            </div>
          </div>
        </li>
        <li
          class="item"
        >
          <img
            alt="images/extrovert-event.jpg"
            src="/images/extrovert-event.jpg"
          />
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <h2>
                Networking for extroverts
              </h2>
              <div
                class="date"
              >
                <time>
                  10 April 2022
                </time>
              </div>
              <div
                class="address"
              >
                <address>
                  My Street 12
10115 Broke City
                </address>
              </div>
            </div>
            <div
              class="actions"
            >
              <a
                href="/events/e3"
              >
                Explore Event
              </a>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <img
          alt="images/coding-event.jpg"
          src="/images/coding-event.jpg"
        />
        <div
          class="content"
        >
          <div
            class="summary"
          >
            <h2>
              Programming for everyone
            </h2>
            <div
              class="date"
            >
              <time>
                12 May 2021
              </time>
            </div>
            <div
              class="address"
            >
              <address>
                Somestreet 25
12345 San Somewhereo
              </address>
            </div>
          </div>
          <div
            class="actions"
          >
            <a
              href="/events/e1"
            >
              Explore Event
            </a>
          </div>
        </div>
      </li>
      <li
        class="item"
      >
        <img
          alt="images/introvert-event.jpg"
          src="/images/introvert-event.jpg"
        />
        <div
          class="content"
        >
          <div
            class="summary"
          >
            <h2>
              Networking for introverts
            </h2>
            <div
              class="date"
            >
              <time>
                30 May 2021
              </time>
            </div>
            <div
              class="address"
            >
              <address>
                New Wall Street 5
98765 New Work
              </address>
            </div>
          </div>
          <div
            class="actions"
          >
            <a
              href="/events/e2"
            >
              Explore Event
            </a>
          </div>
        </div>
      </li>
      <li
        class="item"
      >
        <img
          alt="images/extrovert-event.jpg"
          src="/images/extrovert-event.jpg"
        />
        <div
          class="content"
        >
          <div
            class="summary"
          >
            <h2>
              Networking for extroverts
            </h2>
            <div
              class="date"
            >
              <time>
                10 April 2022
              </time>
            </div>
            <div
              class="address"
            >
              <address>
                My Street 12
10115 Broke City
              </address>
            </div>
          </div>
          <div
            class="actions"
          >
            <a
              href="/events/e3"
            >
              Explore Event
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
